### PR TITLE
Issue #716 Update CI master code to host the docs in artifacts server

### LINKED ISSUE #716: Updated CI master build code to also host the docs in artifacts server
The PR job should also host the docs content in the artifacts server.

The idea is, through CI job, we should:
- Build tar 
- Integrate the docs with the [openshift-docs](https://github.com/openshift/openshift-docs) as instructed [here](https://docs.openshift.org/latest/minishift/developing/writing-docs.html#section-building-openshift-docs)

### PR DESCRIPTION
**Pull request by budhrg**
Fix #716

### CODE REVIEW

**Comment by budhrg** [comment]
retest this please

**Comment by budhrg** [comment]
@hferentschik , this seems to be working as expected. See https://ci.centos.org/job/minishift-pr/681/console job which actually running against the PR job. 

This will only be tested truely when merged.

File: centos_ci.sh [hunk 1]
```diff
@@ -134,24 +134,61 @@ function prepare() {
...
+  cd ..
+  rake build
+  cd ..
+}
+
 function artifacts_upload_on_pr_and_master_trigger() {
   set +x
   # For PR build, GIT_BRANCH is set to branch name other than origin/master
   if [[ "$GIT_BRANCH" = "origin/master" ]]; then
+    install_docs_prerequisite_packages;
```
Comment (hferentschik) [inline centos_ci.sh]: so the docs build is only happening for master right?

Comment (budhrg) [inline centos_ci.sh]: Yes @hferentschik

File: centos_ci.sh [hunk 2]
```diff
@@ -134,24 +134,61 @@ function prepare() {
...
+    build_openshift_origin_docs $(pwd)/docs/build/minishift-adoc.tar;
+
     # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
     mkdir -p minishift/master/$BUILD_NUMBER/
     cp -r out/*-amd64 minishift/master/$BUILD_NUMBER/
+    # Copy the openshift-origin docs
+    cp -r openshift-docs/_preview/openshift-origin minishift/master/$BUILD_NUMBER/
     RSYNC_PASSWORD=$1 rsync -a --delete --relative minishift/master/$BUILD_NUMBER/ minishift@artifacts.ci.centos.org::minishift/
     echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/master/$BUILD_NUMBER ."
+    echo "Minishift docs is hosted at http://artifacts.ci.centos.org/minishift/minishift/master/$BUILD_NUMBER/openshift-origin/latest/minishift/index.html ."
```
Comment (hferentschik) [inline centos_ci.sh]: I think 'openshift-origin' might be confusing. Almost like we are building OpenShift Origin itself. The is actually 'openshift-docs', right?

Comment (budhrg) [inline centos_ci.sh]: ah @hferentschik , make sense Thanks

Comment (budhrg) [inline centos_ci.sh]: Done @hferentschik

File: centos_ci.sh (hunk 1 above)
Comment (hferentschik) [inline centos_ci.sh]: Cool

File: centos_ci.sh [hunk 3]
```diff
@@ -134,24 +134,61 @@ function prepare() {
...
+    build_openshift_origin_docs $(pwd)/docs/build/minishift-adoc.tar;
+
     # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
     mkdir -p minishift/master/$BUILD_NUMBER/
     cp -r out/*-amd64 minishift/master/$BUILD_NUMBER/
+    # Copy the openshift-docs
+    cp -r openshift-docs/_preview/openshift-origin minishift/master/$BUILD_NUMBER/openshift-docs
     RSYNC_PASSWORD=$1 rsync -a --delete --relative minishift/master/$BUILD_NUMBER/ minishift@artifacts.ci.centos.org::minishift/
     echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/master/$BUILD_NUMBER ."
+    echo "Minishift docs is hosted at http://artifacts.ci.centos.org/minishift/minishift/master/$BUILD_NUMBER/openshift-docs/latest/minishift/index.html ."
```
Comment (hferentschik) [inline centos_ci.sh]: 👍

**Pull request merged**